### PR TITLE
chore: move hf examples

### PR DESCRIPTION
**Pull request by garrett361**
## Description
Replaces `examples/integrations/hf_trainer_api` with the examples from `examples/deepspeed/deepspeed_autotune/hf_trainer_api` and also fixes no-op bug in `get_hf_args_with_overwrites`.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: examples/integrations/hf_trainer_api/language_modeling/deepspeed.yaml
Comment (garrett361) [inline examples/integrations/hf_trainer_api/language_modeling/deepspeed.yaml:1]: Just noting that the two `deepspeed.yaml` files still have a `--cache_dir ... ` line which is specific to my Grenoble workflow (makes testing way faster) and needs to be removed eventually.

File: harness/determined/pytorch/dsat/_utils.py [hunk 1]
```diff
@@ -500,16 +500,10 @@ def get_hf_args_with_overwrites(
         logging.info(
             f"{overwrite_key} key not found in hparams, `get_hf_args_with_overwrites` is a no-op"
         )
-        return []
+        return args
```
Comment (tayritenour) [inline harness/determined/pytorch/dsat/_utils.py:503]: ah my bad

**Review by tayritenour — APPROVED** [review]
LGTM

File: harness/determined/pytorch/dsat/_utils.py (hunk 1 above)
Comment (garrett361) [reply harness/determined/pytorch/dsat/_utils.py:503]: I thought I did that.

**Pull request merged**